site-src/guides/infrastructure.md

```diff
@@ -0,0 +1,45 @@
+# Gateway infrastructure labels and annotations
+
+??? success "Standard Channel since v1.2.0"
+
+    The `infrastructure` field is GA and has been part of the Standard Channel since
+    `v1.2.0`. For more information on release channels, refer to our [versioning
+    guide](/concepts/versioning).
+
+Gateway API implementations are responsible for creating the backing
+infrastructure needed to make each Gateway work. For example, implementations
+running in a Kubernetes cluster often create [Services][service] and
+[Deployments][deployment], while cloud-based implementations may create cloud
+load balancer resources. In many cases, it can be helpful to be able to
+propagate labels or annotations to these generated resources.
+
+
+The [`infrastructure` field][infrastructure] on a Gateway allows you to specify
+labels and annotations for the infrastructure created by the Gateway API controller.
+For example, if your Gateway infrastructure is running in-cluster, you can specify
+both Linkerd and Istio injection using the following Gateway configuration, making
+it simpler for the infrastructure to be incorporated into whichever service mesh
+you've installed.
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: meshed-gateway
+  namespace: incoming
+spec:
+  gatewayClassName: meshed-gateway-class
+  listeners:
+  - name: http-listener
+    protocol: HTTP
+    port: 80
+  infrastructure:
+    labels:
+      istio-injection: enabled
+    annotations:
+      linkerd.io/inject: enabled
```

Comment:
```suggestion
    labels:
      implementation-x: feature-foo
    annotations:
      implementation-y: feature-bar
```
I hear and share @arkodg's concerns about the neutrality of Gateway API. I think that by changing the example to something like this and explaining it properly, we don't lose clarity or expressiveness.

Reply:
It's not about an "implementation" (of the Gateway API) here though, it's about applying a label or annotation to the gateway such as some other piece of software in the cluster can consume that. This stanza could be consumed by Envoy Gateway or Emissary. There's no reason why you would need to be configuring either Istio or Linkerd with the Gateway API in this example.

The example as written is self-documenting and clearly explains why you might want to use this feature. Unfortunately, the revised version does not achieve this goal. In general I would suggest a projector prioritise user-first documentation over the perception of a need to be "neutral".

Reply:
I actually had [similar feedback on the blog post PR](https://github.com/kubernetes/website/pull/48270#discussion_r1828558305). Flynn eventually convinced me that mesh injection was _ok_ here, but it's worth pointing out that it's an example that will only work for a subset of implementations that support this feature. 

If on the other hand we stick to more generic labels such as `app`, `env`, etc, they will be equally useful whether they're applied to Kubernetes or Cloud resources.

Reply:
...and in fact @craigbox preserved the language added in response to your feedback, @robscott. 🙁 

I'll stand by my comment below: this feature exists to support implementation-specific things.

Reply:
>...and in fact @craigbox preserved the language added in response to your feedback, @robscott. 🙁

Haha yes, I definitely appreciate that. I still don't _love_ this, but I will agree that it's a reasonable example of how many people will actually use the API. 

> I'll stand by my comment below: this feature exists to support implementation-specific things.

When I pitched this in SIG-Network API review, I highlighted that this could be used to discover resources that were generated for a Gateway or set of Gateways. The idea that this could be used to pass through implementation-specific labels is certainly a likely use of this API, but it does feel rather hacky, so I don't love making that the primary way we showcase the API.